youtube_dl/YoutubeDL.py

```diff
@@ -1237,6 +1237,10 @@ def process_video_result(self, info_dict, download=True):
         if thumbnails and 'thumbnail' not in info_dict:
             info_dict['thumbnail'] = thumbnails[-1]['url']
 
+        if self.params.get('list_thumbnails'):
+            self.list_thumbnails(info_dict)
+            return
```

Comment:
> This can even go before if block at 1237 if we remove references to thumbnail (1905-1908) in list_thumbnails.

if `list_thumbnails` is meant only to be called only by youtube-dl then it will be safer to change the code to:
```python
    def list_thumbnails(self, info_dict):
        thumbnails = info_dict.get('thumbnails')
        if not thumbnails:
            self.to_screen('[info] No thumbnails present for %s' % info_dict['id'])
            return
```
and
```python
info_dict['thumbnails'] = thumbnails = [{'id': '0', 'url': thumbnail}]
```